tasks/go_test.py

```diff
@@ -895,3 +896,67 @@ def format_packages(ctx, impacted_packages):
 
 def normpath(path):  # Normpath with forward slashes to avoid issues on Windows
     return os.path.normpath(path).replace("\\", "/")
+
+
+@task
+def notify_unit_tests(ctx, pipeline_id, pipeline_url, branch_name):
+    from .libs.common.github_api import GithubAPI
+
+    pipeline_id_regex = re.compile(r"pipeline ([0-9]*)")
+
+    no_tests_executed_jobs = process_unit_tests_tarballs(ctx)
+    gh = GithubAPI("DataDog/datadog-agent")
+    pr = gh.get_pr_for_branch(branch_name)[0]
+    # If the branch is not linked to any PR we stop here
+    if pr is None:
+        return
+
+    comment = gh.find_comment(pr.number, "[Fast Unit Tests Report]")
+    if comment is None and len(no_tests_executed_jobs) > 0:
+        msg = create_msg(pipeline_id, pipeline_url, no_tests_executed_jobs)
+        gh.publish_comment(pr.number, msg)
+        return
+
+    previous_comment_pipeline_id = pipeline_id_regex.findall(comment.body)[0]
+    # An older pipeline should not edit a message corresponding to a newer pipeline
+    if previous_comment_pipeline_id > pipeline_id:
```

Comment:
You risk an `IndexError` here
```suggestion
    previous_comment_pipeline_ids = pipeline_id_regex.findall(comment.body)
    # An older pipeline should not edit a message corresponding to a newer pipeline
    if previous_comment_pipeline_ids and previous_comment_pipeline_ids[0] > pipeline_id:
```

Reply:
Thanks! Updated